Description: Add --with-system-json-c flag to build against system libjson-c.
Author: Ludovico Cavedon <cavedon@debian.org>

Index: ndpi/example/Makefile.am
===================================================================
--- ndpi.orig/example/Makefile.am
+++ ndpi/example/Makefile.am
@@ -1,9 +1,9 @@
 bin_PROGRAMS = ndpiReader
 
-AM_CPPFLAGS = -I$(top_srcdir)/src/include @PCAP_INC@
+AM_CPPFLAGS = -I$(top_srcdir)/src/include @PCAP_INC@ @JSON_C_CFLAGS@
 AM_CFLAGS = @PTHREAD_CFLAGS@ # --coverage
 
-LDADD = $(top_builddir)/src/lib/libndpi.la @JSON_C_LIB@ @PTHREAD_LIBS@ @PCAP_LIB@ @DL_LIB@ -lm
+LDADD = $(top_builddir)/src/lib/libndpi.la @JSON_C_LIBS@ @PTHREAD_LIBS@ @PCAP_LIB@ @DL_LIB@ -lm
 AM_LDFLAGS = -static @DL_LIB@
 
 ndpiReader_SOURCES = ndpiReader.c ndpi_util.c ndpi_util.h uthash.h
Index: ndpi/configure.seed
===================================================================
--- ndpi.orig/configure.seed
+++ ndpi/configure.seed
@@ -79,35 +79,20 @@ AC_ARG_ENABLE([json-c],
     AS_HELP_STRING([--disable-json-c], [Disable json-c support]))
 
 AS_IF([test "x$enable_json_c" != "xno"], [
-       PKG_CONFIG_PATH=/usr/local/share/pkgconfig:$PKG_CONFIG_PATH
-       pkg-config --exists json-c
-       AS_IF([test "$?" == "0"],
-             [
-              CFLAGS="$CFLAGS $(pkg-config --cflags json-c)"
-              LDFLAGS="$LDFLAGS $(pkg-config --libs json-c)"
-              AC_CHECK_LIB(json-c, json_object_new_object, AC_DEFINE_UNQUOTED(HAVE_JSON_C, 1, [The JSON-C library is present]))
-             ],
-             [
-		JSONC_HOME="$HOME/json-c"
-	          if test -d "$JSONC_HOME"; then :
-		        CFLAGS="$CFLAGS -I $JSONC_HOME"
-			LDFLAGS="$LDFLAGS $JSONC_HOME/.libs/libjson-c.a"
-    		        AC_MSG_RESULT([Found json-c in $JSONC_HOME])
-		 	AC_DEFINE_UNQUOTED(HAVE_JSON_C, 1, [The JSON-C library is present])
-		  fi
-	     ])
-       ])
+      PKG_CHECK_MODULES([JSON_C], [json-c], [AC_DEFINE([HAVE_JSON_C], [1], [Use libjson-c])])
+])
 
 AC_CHECK_LIB(pthread, pthread_setaffinity_np, AC_DEFINE_UNQUOTED(HAVE_PTHREAD_SETAFFINITY_NP, 1, [libc has pthread_setaffinity_np]))
 
 AC_CONFIG_FILES([Makefile src/lib/Makefile example/Makefile tests/Makefile libndpi.pc src/include/ndpi_define.h])
 AC_CONFIG_HEADERS(config.h)
+AC_SUBST(JSON_C_CFLAGS)
+AC_SUBST(JSON_C_LIBS)
 AC_SUBST(GIT_RELEASE)
 AC_SUBST(NDPI_MAJOR)
 AC_SUBST(NDPI_MINOR)
 AC_SUBST(NDPI_PATCH)
 AC_SUBST(SVN_DATE)
-AC_SUBST(JSON_C_LIB)
 AC_SUBST(PCAP_INC)
 AC_SUBST(PCAP_LIB)
 AC_SUBST(DL_LIB)
